react/components/BaseModal.tsx

```diff
@@ -42,14 +42,17 @@ export default function BaseModal(props: Props) {
   } = props
 
   const [exited, setExited] = useState(true)
+  const [prevOpen, setPrevOpen] = useState(open)
 
-  useEffect(() => {
+  if (open !== prevOpen) {
+    setPrevOpen(open)
     if (open) {
-      document.body.classList.add('overflow-y-hidden')
+      // This can't be overflow-y-hidden because of some problem with safari
+      document.body.classList.add('overflow-hidden')
```

Comment:
Use a custom CSS Modules class to do this so it's easier to track down which component is doing this side-effect. Example:

`vtex-modal-layout-0-x-overflow-hidden`